package/version.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -xeuo pipefail
+
+notif() { echo "== $@" >&2 ; }
+fatal() { echo "[FATAL] $@" ; exit 1 ; }
+
+version_file="package/version"
+
+version_bump() {
+    local version release_commit version_major version_minor version_patch new_version current_version current_version_major current_version_minor current_version_patch
+    version="$1" ; shift
+    version_major="$(echo ${version} | cut --delimiter '.' --field 1)"
+    version_minor="$(echo ${version} | cut --delimiter '.' --field 2)"
+    version_patch="$(echo ${version} | cut --delimiter '.' --field 3)"
+    current_version="$(cat ${version_file})"
+    current_version_major="$(echo ${current_version} | cut --delimiter '.' --field 1)"
+    current_version_minor="$(echo ${current_version} | cut --delimiter '.' --field 2)"
+    current_version_patch="$(echo ${current_version} | cut --delimiter '.' --field 3)"
+    new_version="${version}"
+    if [[ "${version_major}" == "${current_version_major}" ]] && [[ "${version_minor}" == "${current_version_minor}" ]]; then
+        new_version="${version_major}.${version_minor}.$((version_patch + 1))"
+    fi
+    # If the file being commited increases the major or minor, then take that version as it is
+    if [[ "${version_major}" < "${current_version_major}" ]] || [[ "${version_minor}" < "${current_version_minor}" ]]; then
+        new_version="${current_version}"
+    fi
```

Comment:
@F-WRunTime I want us to have the ability to increase the major and minor by updating `package/version`, does this seem reasonable?

Reply:
Yes this looks good. If you want to rev the major version this is a manual step on your part, the version.sh script will ignore rev'ing the minor version in this scenario.